### PR TITLE
Solution: Max Area of Island

### PR DESCRIPTION
**Pull request by 7mada123**

### CODE REVIEW

File: src/com/coding/patterns/graphs/MaxAreaofIsland.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package com.coding.patterns.graphs;
```
Comment (ndesai15) [inline src/com/coding/patterns/graphs/MaxAreaofIsland.java]: Can you please move this problem under dfs pattern package?

File: src/com/coding/patterns/graphs/MaxAreaofIsland.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        }));
+
+        // Output 0
+        System.out.println(maxAreaOfIsland(new int[][] { { 0, 0, 0, 0, 0, 0, 0, 0 } }));
+    }
+
+    static public int maxAreaOfIsland(int[][] grid) {
+        int max = 0;
+
+        for (int i = 0; i < grid.length; i++)
```
Comment (ndesai15) [inline src/com/coding/patterns/graphs/MaxAreaofIsland.java]: To maintain standards across repository, let's assign `grid. length` to `rows` variable and use that variable.

File: src/com/coding/patterns/graphs/MaxAreaofIsland.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            for (int j = 0; j < grid[i].length; j++)
```
Comment (ndesai15) [inline src/com/coding/patterns/graphs/MaxAreaofIsland.java]: Let's declare this variable outside of for loop and call it `cols`.

File: src/com/coding/patterns/graphs/MaxAreaofIsland.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                if (grid[i][j] == 1)
+                    max = Math.max(max, getSize(grid, i, j));
+
+        return max;
+    }
+
+    static int getSize(int[][] grid, int i, int j) {
```
Comment (ndesai15) [inline src/com/coding/patterns/graphs/MaxAreaofIsland.java]: Change this method to `dfs()`

**Review by 7mada123 — COMMENTED** [review]
Moved to dfs pattern package

Changed names to the repository standards (rows, cols, res, dfs)

**Comment by ndesai15** [comment]
Thank You for your contribution to this repository. I am going to merge this change to master. If you would like to contribute to open issues, please check open issues with help wanted tags.

Thank You
~Neel Desai

**Pull request merged**